src/pages/Account/Venue/VenueMapEditor/Container.tsx

```diff
@@ -0,0 +1,242 @@
+  safeZone: safeZone100,
+  interactive,
+  rounded,
+  lockAspectRatio,
+  isSaving,
+}) => {
+  // For historical reasons, the icons coming in have all dimensions specified
+  // as percentages in the 0 to 100 range rather than 0 to 1. Convert them here
+  // to make further calculations easier
+  const iconsMapPerc = useMemo(
+    () =>
+      iconsMap100.map((icon) => ({
+        ...icon,
+        width: icon.width / 100,
+        height: icon.height / 100,
+        top: icon.top / 100,
+        left: icon.left / 100,
+      })),
+    [iconsMap100]
+  );
+  const [boxes, setBoxes] = useState<SubVenueIcon[]>(iconsMapPerc);
+  const [imageDims, setImageDims] = useState<Dimensions>();
+  const [dragBoxId, setDragBoxId] = useState<number>(0);
+  const wrapperRef = useRef(null);
+
+  // For historical reasons, the safe zone passed in is 0 to 100 rather than
+  // 0 to 1, convert it to make calculations later easier.
+  const safeZonePerc = useMemo(
+    () => ({
+      width: safeZone100.width / 100,
+      height: safeZone100.height / 100,
+    }),
+    [safeZone100]
+  );
+
+  const onRoomResize = useCallback((width, height) => {
+    setImageDims({ width, height });
+  }, []);
+
+  const { ref } = useResizeDetector({
+    onResize: onRoomResize,
+    targetRef: wrapperRef,
+  });
+
+  const setDragItemId = useCallback((id: number) => {
+    setDragBoxId(id);
+  }, []);
+
+  // trigger the parent callback on boxes change (as a result of movement)
+  useEffect(() => {
+    if (!imageDims || !imageDims.height || !imageDims.width) return;
+
+    const convertDisplayedCoordToIntrinsic = (
+      val: number,
+      dimension: keyof typeof imageDims,
+      safeZoneDimension: number
+    ) => {
+      const imageDim = imageDims[dimension];
+      return (
+        100 *
+        ((val - imageDim * ((1 - safeZoneDimension) / 2)) /
+          (imageDim * safeZoneDimension))
+      );
+    };
+    //need to return the unscaled values
+    const unscaledBoxes = boxes.map(
+      (box) =>
+        ({
+          ...box,
+          width: (100 * box.width) / (imageDims.width * safeZonePerc.width),
+          height: (100 * box.height) / (imageDims.height * safeZonePerc.height),
+          top: convertDisplayedCoordToIntrinsic(
+            box.top,
+            "height",
+            safeZonePerc.height
+          ),
+          left: convertDisplayedCoordToIntrinsic(
+            box.left,
+            "width",
+            safeZonePerc.width
+          ),
+        } as SubVenueIcon)
+    );
+
+    onChange && onChange(unscaledBoxes);
+  }, [boxes, onChange, imageDims, safeZonePerc.width, safeZonePerc.height]);
+
+  useMemo(() => {
+    if (!imageDims || !imageDims.width || !imageDims.height) return;
+
+    const copy = iconsMapPerc.map(
+      (mapIcon) =>
+        ({
+          ...mapIcon,
+          width: imageDims.width * mapIcon.width * safeZonePerc.width,
+          height: imageDims.height * mapIcon.height * safeZonePerc.height,
+          top:
```

Comment:
I think there should be an explanation comment line or two here and above about the need to multiply or divide something by a factor. Also, since this is a function inside `.map()` it might be easy to extract it outside of this component